worker/echo/main.go

```diff
@@ -15,19 +16,40 @@ import (
 )
 
 var sleepTime time.Duration
+var cancelChan map[string]chan struct{}
+var chanMutex = &sync.RWMutex{}
 
 // echo opens a new dbus connection and calls the
 // com.redhat.Yggdrasil1.Dispatcher1.Transmit method, returning the metadata and
-// data it received.
+// data it received. Cancelled messages will only be handled if the worker is running
+// in the slow mode.
 func echo(w *worker.Worker, addr string, id string, responseTo string, metadata map[string]string, data []byte) error {
 	if err := w.EmitEvent(ipc.WorkerEventNameWorking, fmt.Sprintf("echoing %v", data)); err != nil {
 		return fmt.Errorf("cannot call EmitEvent: %w", err)
 	}
 
 	// Sleep time between receiving the message and sending it
 	if sleepTime > 0 {
-		log.Infof("sleeping: %v", sleepTime)
+		// Setting the channel to handle cancelation
+		chanMutex.Lock()
+		cancelChan[id] = make(chan struct{})
```

Comment:
Isn't too late for creating this channel? Shouldn't we create such channel in main function of this worker before worker is created? It looks like that it will be always necessary to create such channel. Shouldn't be such channel  integrated part of `Worker` structure, and shouldn't we call all of this in `NewWorker()`?

Reply:
No, we create the map of channels in the main function, which contains the MessageID as the key and the channel as the value. 
The channels of that map should be created when a message arrives at the echo worker and is running with sleep time set. That is a way the worker can control the pending jobs that are running and can be canceled.

This code is the specific implementation of the cancel message for the echo worker. But not all workers will implement such things neither in the same manner.

> Shouldn't be such channel integrated part of the Worker structure, and shouldn't we call all of this in NewWorker()?

Channels are specific to Go. I would try to make the worker the more generally possible to its implementation with libyggd (the C library).

Reply:
Ah, I can see. The `cancelChan` is created here: https://github.com/RedHatInsights/yggdrasil/pull/122/files#diff-a3f7add2ec318476155f7f939e4242b221a6611a4e50181d2e96fbd13a9ff460R92. Maybe the name `cancelChan` is little bit misleading. What about something like `mapOfCancelChannels`?

Reply:
Make sense if it is not clear. Changing it.